presto-main/src/test/java/com/facebook/presto/cost/TestCostCalculator.java

```diff
@@ -366,17 +366,18 @@ private ValuesNode values(int planNodeId, String... symbols)
     private class FixedLookup
             implements Lookup
     {
-        private Map<PlanNodeId, PlanNodeCostEstimate> costs;
-        private Map<PlanNodeId, PlanNodeStatsEstimate> stats;
+        private Function<String, PlanNodeCostEstimate> costs;
+        private Function<String, PlanNodeStatsEstimate> stats;
+
+        public FixedLookup(Function<String, PlanNodeCostEstimate> costs, Function<String, PlanNodeStatsEstimate> stats)
+        {
+            this.costs = costs;
+            this.stats = stats;
+        }
 
         public FixedLookup(Map<String, PlanNodeCostEstimate> costs, Map<String, PlanNodeStatsEstimate> stats)
         {
-            this.costs = costs.entrySet()
-                    .stream()
-                    .collect(toMap(entry -> new PlanNodeId(entry.getKey()), Map.Entry::getValue));
-            this.stats = stats.entrySet()
-                    .stream()
-                    .collect(toMap(entry -> new PlanNodeId(entry.getKey()), Map.Entry::getValue));
+            this(costs::get, stats::get);
```

Comment:
Are you sure that previous implementation wasn't copying the data purposely?
If we don't need copy nor PlanNodeId instead of Strings then we can just assign `costs` and `stats` to local reference and use `get` on them in `getStats` and `getCosts` instead adding a layer of functional-hocus-pocus. Can't we?
Unless of course you are afraid that someone may try to modify this map in other methods instead just reading from it.